1.2.0 alpha.3 changelog (#3671)
mapbox/mapbox-navigation-android: CHANGELOG.md

Mapbox welcomes participation and contributions from everyone.

### Mapbox Navigation SDK 1.2.0-alpha.3 - October 21, 2020

Visit our [1.0.0 migration guide](https://github.com/mapbox/mapbox-navigation-android/wiki/1.0-Navigation-SDK-Migration-Guide) and the [documentation pages](http://docs.mapbox.com/android/navigation/overview) for more information.

### Changelog

#### Features

> * Update the algorithm to calculate the vanishing point of the route to use EPSG:3857 projection instead of turf measurements. The precision is now spot on, even on extremely long routes, it even greatly outperforms the precision of the over zoomed map tiles. It also improves the performance of the `MapRouteLine` [#3661](https://github.com/mapbox/mapbox-navigation-android/pull/3661)

Reviewer: ```suggestion
* Update the algorithm to calculate the vanishing point of the route to use EPSG:3857 projection instead of turf measurements. This greatly increases the precision, even on extremely long routes. It also improves the performance of the `MapRouteLine`. [#3661](https://github.com/mapbox/mapbox-navigation-android/pull/3661)
```

Also, I'd title it _Improvements_, not _Features_.

Author: done

#### Bug fixes
* Assign `currentHorizon`, `currentType` and `currentPosition` from the main thread to avoid synchronization issues [#3653](https://github.com/mapbox/mapbox-navigation-android/pull/3653)
* Keep the selected route as primary after map style has been changed [#3664](https://github.com/mapbox/mapbox-navigation-android/pull/3664)

> * Don't obfuscate `ReplayDataSource` to prevent a crash in `ReplayHistoryActivity` [#3646](https://github.com/mapbox/mapbox-navigation-android/pull/3646)

Reviewer: Let's remove this entry, it's a test-app change.

Author: done

#### Other changes

> * Bump mapboxCommonNative version to 7.1.1 [#3650](https://github.com/mapbox/mapbox-navigation-android/pull/3650)

Reviewer: Let's remove this note, we're listing the version at the top already.

Author: done

> * Add Electronic Horizon Builder tests [#3655](https://github.com/mapbox/mapbox-navigation-android/pull/3655)

Reviewer: This change is not impacting end-users, let's remove.

Author: done

> * Override commonOkHttp dependency with 7.1.1 [#3657](https://github.com/mapbox/mapbox-navigation-android/pull/3657)

Reviewer: Same as above.

Author: done

> * Remove dependency of UI SDK on the modular notification classes [#3665](https://github.com/mapbox/mapbox-navigation-android/pull/3665)

Reviewer: ```suggestion
* Remove dependency of UI SDK on the modular notification classes. This fixes an issue where the `TripNotification` implementation couldn't be replaced with a custom one. [#3665](https://github.com/mapbox/mapbox-navigation-android/pull/3665)
```

Author: done

> * Bump mapboxSdkServices version to 5.8.0-beta.2 [#3668](https://github.com/mapbox/mapbox-navigation-android/pull/3668)

Reviewer: Same as above.

Author: done
